### ISSUE TITLE
Add support for NC 23

### ISSUE THREAD

**Issue by jurgenhaas** [nextcloud/user_saml#562]
The app store at https://apps.nextcloud.com/apps/user_saml doesn't show support for NC 23 yet, are you planning to support that platform?

**Comment by acsfer** [comment]
This should be the case already 🤔 
https://github.com/nextcloud/user_saml/blob/master/appinfo/info.xml#L36

**Comment by l00ptr** [comment]
nope the latest release was done before the commit increasing the max version :)

**Comment by fwininger** [comment]
Someone can release a new version ?

**Comment by acsfer** [comment]
@nickvergessen possible to Bump version here, as https://github.com/nextcloud/user_saml/pull/560/commits/aea4ca111b058d3d70be0fb1e2c1d489206fc2d4 has been implemented?

**Comment by nickvergessen** [comment]
I'm a bot for version updates, I can not judge the content of the apps. I do ~120 such commit for different apps.
Find contributors/maintainers at https://github.com/nextcloud/user_saml/graphs/contributors (and ignore me)

**Comment by fschrempf** [comment]
@rullzer As you seem to have done the previous releases, are you able to come up with a release for NC23? Thanks!

**Comment by Buri** [comment]
Any update on this?

**Comment by peacepenguin** [comment]
Can someone just tag a new release and get on with this? Nextcloud 23 is looking a little sad without proper SSO/SAML support that's been really solid for many years up till this release. I thought "Featured" apps in the store were vetted and confirmed to be working before a major release? or was this a one-off oversight?

**Comment by fschrempf** [comment]
> I thought "Featured" apps in the store were vetted and confirmed to be working before a major release?

Unfortunately, it seems like this is not the case. One would expect this, but there are other "featured" apps (e. g. groupfolders) that have "missed" major releases and where it took several weeks for the app developers to catch up. I would really like to see some plan/strategy to prevent this, but I also see how the current structure makes it rather difficult for the developers to handle this.

**Comment by peacepenguin** [comment]
Perhaps core functionality like SAML/SSO shouldn't be an app at all? This should simply be integrated into the server code itself. Just like FIDO2 and two factor auth have become integrated now.

Especially considering the saml app lives under the official nextcloud organization in github. That gives the illusion that its not just a 3rd party app, but actually maintained by Nextcloud as a company/organization.

Bizarre that nextcloud advertises saml/sso as a core feature, but then it's actually "just an app" that might not even work at all if you happened to deploy a new server today, based in the current download links, you would be surprised to see SAML not available.

**Comment by fschrempf** [comment]
@peacepenguin I fully agree. We were having a similar discussion about integration into the core for the groupfolders app here: https://github.com/nextcloud/groupfolders/issues/1215#issuecomment-812594556.

And here is the same situation (release delayed) for the groupfolders app and NC22: https://github.com/nextcloud/groupfolders/issues/1598.

Just for reference. Let's not go too far  off topic here... ;)

**Comment by benhartwich** [comment]
Has anyone tried the upgrade with the current version? Is there any problem, which prohibits updating although it is not officially supported?

**Comment by peacepenguin** [comment]
@benhartwich yes I've enabled SAML manually again as an "Untested app" from the app store after the upgrade to 23 disabled it as incompatible, seems to work fine for us in our deployment.

**Comment by keithjnewman** [comment]
Do we have an ETA for NC 23 support yet?

**Comment by rhclayto** [comment]
My whole user log-in system depends on this app. Upgrading to 23 is blocked by this app. Please update.

**Comment by Cludch** [comment]
The workaround from @peacepenguin works fine for the time-being. You can also use the occ cli to re-enable the app.
This will at least allow you to log-in again.

**Comment by rhclayto** [comment]
> The workaround from @peacepenguin works fine for the time-being. You can also use the occ cli to re-enable the app. This will at least allow you to log-in again.

Cli enabling worked, but note to others, you have to manually edit the app's info.xml file:

`<nextcloud min-version="21" max-version="22" />`

must be changed to

`<nextcloud min-version="21" max-version="23" />`

**Comment by faust64** [comment]
Code in master seems to be ready for NC23, since June 29th
https://github.com/nextcloud/user_saml/commit/c36dc6dc841ebd90f62d7ace98c9de7775028e8d

Any reason to hold on a release?

**Comment by wwwwiii** [comment]
Any update regarding this?
I hope future supported wont be dropped...

Thanks!

**Comment by blizzz** [comment]
> Any update regarding this? 

Please have an eye on the app store.

**Comment by luclu** [comment]
Unfortunately and suprisingly I found Nextcloud 23.0.1 not working with the current version of user_saml.
```
occ app:enable user_saml
--> App "SSO & SAML authentication" cannot be installed because it is not compatible with this version of the server.
```
Is there anything I can do to help - like incrementing the supported version field?

**Comment by josephcsible** [comment]
@luclu Use `occ app:enable -f user_saml` to do that.

**Comment by RayChimera** [comment]
Is there at least a rough estimate when the app will be available? Like ~week, ~month, ~year?

**Comment by edmundlaugasson** [comment]
[Proposed to add that feature into Nextcloud core](https://github.com/nextcloud/server/issues/31233), so we won't need any additional app anymore.

**Comment by small1** [comment]
> [Proposed to add that feature into Nextcloud core](https://github.com/nextcloud/server/issues/31233), so we won't need any additional app anymore.

This is happening to a few apps now and then that is more or less considered enterprise.  They will be ready when they release the enterprise part of 23. And that can take a while. Groupfolders had the same last year. 

And if you have a big enough environment ... perhaps going to 23 might be a bit early still.

**Comment by PVince81** [comment]
there will be upcoming config format changes: https://github.com/nextcloud/user_saml/pull/558

after that we can mark the app as compatible with 23

the PR is reaching completion and needs further testing, feel free to help out

**Comment by AndyScherzinger** [comment]
App has been released now, see https://apps.nextcloud.com/apps/user_saml / https://github.com/nextcloud/user_saml/releases/tag/v5.0.0
